sentry-spring/src/main/java/io/sentry/spring/graphql/SentryDataFetcherExceptionResolverAdapter.java

```diff
@@ -0,0 +1,43 @@
+package io.sentry.spring.graphql;
+
+import graphql.GraphQLError;
+import graphql.execution.DataFetcherExceptionHandlerResult;
+import graphql.schema.DataFetchingEnvironment;
+import io.sentry.graphql.SentryGraphqlExceptionHandler;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.graphql.execution.DataFetcherExceptionResolverAdapter;
+
+public final class SentryDataFetcherExceptionResolverAdapter
+    extends DataFetcherExceptionResolverAdapter {
+  private final @NotNull SentryGraphqlExceptionHandler handler;
+
+  public SentryDataFetcherExceptionResolverAdapter() {
+    this.handler = new SentryGraphqlExceptionHandler(null);
+  }
+
+  @Override
+  public boolean isThreadLocalContextAware() {
+    return true;
+  }
+
+  @Override
+  protected @Nullable GraphQLError resolveToSingleError(Throwable ex, DataFetchingEnvironment env) {
```

Comment:
Is there a reason to resolve to a single error instead of all?

Reply:
haven't checked when which of those is called, i assumed it was up to the framework calling it depending on what it wants

Reply:
To keep code reusable it's easiest to to keep the code as is here. `DataFetcherExceptionHandlerResult` may contain multiple errors (`List<GraphQLError> errors`) and that's what we used to have in existing code for raw `graphql-java` and Netflix DGS where a delegate can be used with our `SentryDataFetcherExceptionHandler`.